### DISCUSSION TITLE
External SPI Flash Memory

### DISCUSSION THREAD

**Discussion by gabsi-w** [micropython/micropython#15644] [STM32 / Pyboard] [answered]
Hi, I am new with the micropython and I am currently adapting it to a custom hardware, based on a STM32F746NE chip. At the moment I am trying to make the external flash chips to work. The one connected to the QSPI I already got to work but now I don't know how to do it with the one connected to the SPI interface. What can I possiply do? Is only the MBOOT option available? My custom hardware does not use USB, only SWD, does the MBOOT wotk in thi case?

**Comment by andrewleech** [comment, marked as answer]
If you look at the PYBD boards bdev.c file and mpconfigboard.h they use two external flash chips, one on QSPI which can be used for the main firmware executable code (supports memory mapped use), the second is connected with SoftQSPI and can be used for the filesystem data storage. https://github.com/micropython/micropython/blob/master/ports/stm32/boards/PYBD_SF2/bdev.c

If your second one is wired to hardware spi you can use that instead of softqspi configuration eg. https://github.com/micropython/micropython/blob/a6fa85d8f96897863546423b2bbfc2c24237040f/ports/stm32/boards/ADAFRUIT_F405_EXPRESS/bdev.c#L4

If you only program your board via SWD you likely wont need to use / build with mboot support at all, unless you want OTA support in future to load firmware from filesystem (called fsload in mboot).

**Reply by gabsi-w** [reply]
Thanks for the reply, your answer helped me a lot to make the SPI memory work. Now both of them are working, but separately. 

My problem now is getting the two to work together and make sure they are working. I am now trying to do this and using pyb.info() to check the memory available, but I am always getting 8 MB instead of 16 MB. How can I get to have both defined and recognized by micropython if I can have only one MICROPY_HW_BDEV_SPIFLASH defined, for example?
